api/src/main/java/org/openmrs/module/ugandaemrreports/reporting/dataset/definition/SharedDataDefintion.java

```diff
@@ -36,4 +41,29 @@ public DataDefinition definition(String name, Concept concept) {
         obsForPersonDataDefinition.setWhich(TimeQualifier.LAST);
         return obsForPersonDataDefinition;
     }
+    public DataDefinition villageParish(){
+        CalculationDataDefinition cdf =new CalculationDataDefinition("village+parish", new PersonAddressCalculation());
+        cdf.addParameter(new Parameter("onDate", "On Date", Date.class));
+        return cdf;
+    }
+    public DataDefinition age(Integer lower, Integer upper) {
+        CalculationDataDefinition cdf = new CalculationDataDefinition("Age-"+lower+"-"+upper+"yrs", new AgeLimitCalculation());
+        cdf.addCalculationParameter("lowerLimit", lower);
+        cdf.addCalculationParameter("upperLimit", upper);
+        cdf.addParameter(new Parameter("onDate", "On Date", Date.class));
+        return cdf;
+    }
+    public DataDefinition whoCd4Vl(String q, String a){
```

Comment:
@solemabrothers I think also add the Alternate and Second Alternate Phone number values